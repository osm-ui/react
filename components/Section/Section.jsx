import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styled from 'styled-components';


const StyledSection = styled.section`
    color: ${props => props.theme.color};
    background-color: ${props => props.theme.backgroundColor};

    & > * {
        opacity: ${props => (props.dimmed ? 0.2 : 1)};
    }

    &.app-canvas {
        position: absolute;
        top: 0;
        right: 0;
        bottom: 0;
        left: 0;
        width: 100%;
        height: 100%;
        overflow: hidden;
    }
`;


const Section = ({ appCanvas, className, ...props }) => {
    const classes = classnames(className, {
        'app-canvas': appCanvas,
    });

    return (
        <StyledSection className={classes} {...props} />
    );
};


Section.propTypes = {
    appCanvas: PropTypes.bool,
    dimmed: PropTypes.bool,
    className: PropTypes.string,
};

Section.defaultProps = {
    appCanvas: false,
    dimmed: false,
    className: '',
};

Section.displayName = 'Section';

export default Section;
