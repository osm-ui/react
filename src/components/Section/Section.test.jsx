import React from 'react';
import {
  snapshotWithElementChildren,
  snapshotWithTextChildren
} from '../../helpers/tests';
import Section from './';

describe('When using snapshots', () => {
  it('Should render with an element children', () =>
    snapshotWithElementChildren(Section));

  it('Should render with a text children', () =>
    snapshotWithTextChildren(Section));
});

describe('When using snapshots with the dimmed prop', () => {
  it('Should render with an element children', () =>
    snapshotWithElementChildren(Section, { dimmed: true }));

  it('Should render with a text children', () =>
    snapshotWithTextChildren(Section, { dimmed: true }));
});
