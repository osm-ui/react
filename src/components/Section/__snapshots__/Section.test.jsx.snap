// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When using snapshots Should render with a text children 1`] = `
<span
  className="sc-bdVaJa lbJuWU"
>
  <section
    className="sc-bwzfXH bWibny"
  >
    This is a text
  </section>
</span>
`;

exports[`When using snapshots Should render with an element children 1`] = `
<span
  className="sc-bdVaJa lbJuWU"
>
  <section
    className="sc-bwzfXH bWibny"
  >
    <div />
  </section>
</span>
`;

exports[`When using snapshots with the dimmed prop Should render with a text children 1`] = `
<span
  className="sc-bdVaJa lbJuWU"
>
  <section
    className="sc-bwzfXH cbGbWf"
  >
    This is a text
  </section>
</span>
`;

exports[`When using snapshots with the dimmed prop Should render with an element children 1`] = `
<span
  className="sc-bdVaJa lbJuWU"
>
  <section
    className="sc-bwzfXH cbGbWf"
  >
    <div />
  </section>
</span>
`;
